test/currency.js

```diff
@@ -0,0 +1,126 @@
+    ccAddress = result.logs[0].args.token
+    const cc = await ColuLocalCurrency.at(ccAddress)
+
+    mmAddress = await currencyFactory.getMarketMakerAddressFromToken(ccAddress)
+    const mm = await EllipseMarketMaker.at(mmAddress)
+
+    await currencyFactory.openMarket(ccAddress)
+
+    osseus = await Osseus.get()
+    osseus.cwd = osseus.cwd || cwd
+    osseus.db_models = osseus.db_models || {
+      currency: require(path.join(cwd, 'modules/db/models/currency'))(osseus.mongo),
+      community: require(path.join(cwd, 'modules/db/models/community'))(osseus.mongo)
+    }
+    osseus.lib = osseus.lib || {
+      Currency: require(path.join(cwd, 'modules/lib/models/currency'))(osseus),
+      Community: require(path.join(cwd, 'modules/lib/models/community'))(osseus)
+    }
+  })
+
+  beforeEach(function () {
+    // console.log('--- beforeEach [currency] ---')
+    Object.keys(osseus.db_models).forEach(model => {
+      // console.log(`model: ${model}`)
+      osseus.db_models[model].getModel().remove({}, () => {})
+    })
+  })
+
+  it('should create a currency', async () => {
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    validateCurrency(currency)
+  })
+
+  it('should not create a currency with same address', async () => {
+    await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI).catch(err => {
+      expect(err).not.to.be.undefined
+    })
+    expect(currency).to.be.undefined
+  })
+
+  it('should get currency (by id)', async () => {
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    currency = await osseus.db_models.currency.getById(currency.id)
+    validateCurrency(currency)
+  })
+
+  it('should get currency (by cc address)', async () => {
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    currency = await osseus.db_models.currency.getByCurrencyAddress(currency.ccAddress)
+    validateCurrency(currency)
+  })
+
+  it('should get currency (by mm address)', async () => {
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    currency = await osseus.db_models.currency.getByMarketMakerAddress(currency.mmAddress)
+    validateCurrency(currency)
+  })
+
+  it('should get error if currency not found (by id)', async () => {
+    let fakeId = '123abc'
+    let currency = await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    currency = await osseus.db_models.currency.getById(fakeId).catch(err => {
+      expect(err).not.to.be.undefined
+    })
+    expect(currency).to.be.undefined
+  })
+
+  it('should get all currencies', async () => {
+    await osseus.lib.Currency.create(ccAddress, mmAddress, ccABI, mmABI)
+    let currencies = await osseus.db_models.currency.getAll()
```

Comment:
and here
maybe try to make multiple currencies as you do with communities also.